src/database/schema.zmodel

```diff
@@ -797,3 +804,178 @@ model LastKeycloakUpdate {
 
   @@map("last_keycloak_update")
 }
+
+
+
+// SHOP SYSTEM
+
+enum ShoppableType {
+  TICKET
+}
+
+// Base entity to be inherited by actual purchasables
+model Shoppable {
+  id String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  title String
+  titleEn String?
+  description String
+  descriptionEn String?
+  type ShoppableType
+
+  authorId String @db.Uuid @deny("update", true) // never allow updating wh created the shoppable
+  author Member @relation(fields: [authorId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+
+  price Int
+  availableFrom DateTime @map("available_from") @db.Timestamptz(6) @default(now())
```

Comment:
Maybe consider having this nullable. This would allow for authors to create drafts for items before they know when they are to be published.

Maybe that could be changed later, though.

Reply:
Also, items that are never meant to be sold? Like café tickets would be consumables, but they need a shoppable as well, but it should never be published!

Reply:
I think this is a different issue which could be handled in a future PR. Perhaps together with "ItemAccessPolicies" where we could se custom policies for who can buy a Shoppable. Café tickets could then have the policy "webhsop:item:not_visibile" or something.

I don't want to increase the scope of these PRs unecessarily